File: OTCDerivativesCalculatorModule/Project_Cpp/lib_static/FpmlSerialized/GenClass/fpml-asset-5-4/PendingPayment.cpp
// PendingPayment.cpp 
#include "PendingPayment.hpp"
#ifdef ConsolePrint
    #include <iostream>
#endif 

namespace FpmlSerialized {

PendingPayment::PendingPayment(TiXmlNode* xmlNode)
: PaymentBase(xmlNode)
{
    #ifdef ConsolePrint
        std::string initialtap_ = FileManager::instance().tap_;
        FileManager::instance().tap_.append("   ");
    #endif 
   //paymentDateNode ----------------------------------------------------------------------------------------------------------------------
   TiXmlElement* paymentDateNode = xmlNode->FirstChildElement("paymentDate");

   if(paymentDateNode){paymentDateIsNull_ = false;}
   else{paymentDateIsNull_ = true;}

   #ifdef ConsolePrint
      FileManager::instance().outFile_ << FileManager::instance().tap_.c_str() << "- paymentDateNode , address : " << paymentDateNode << std::endl;
   #endif
   if(paymentDateNode)
   {
      if(paymentDateNode->Attribute("href") || paymentDateNode->Attribute("id"))
      {
          if(paymentDateNode->Attribute("id"))
          {
             paymentDateIDRef_ = paymentDateNode->Attribute("id");
             paymentDate_ = boost::shared_ptr<XsdTypeDate>(new XsdTypeDate(paymentDateNode));
             paymentDate_->setID(paymentDateIDRef_);
             IDManager::instance().setID(paymentDateIDRef_,paymentDate_);
          }
          else if(paymentDateNode->Attribute("href")) { paymentDateIDRef_ = paymentDateNode->Attribute("href");}
          else { QL_FAIL("id or href error"); }
      }
      else { paymentDate_ = boost::shared_ptr<XsdTypeDate>(new XsdTypeDate(paymentDateNode));}
   }

   //amountNode ----------------------------------------------------------------------------------------------------------------------
   TiXmlElement* amountNode = xmlNode->FirstChildElement("amount");

   if(amountNode){amountIsNull_ = false;}
   else{amountIsNull_ = true;}

   #ifdef ConsolePrint
      FileManager::instance().outFile_ << FileManager::instance().tap_.c_str() << "- amountNode , address : " << amountNode << std::endl;
   #endif
   if(amountNode)
   {
      if(amountNode->Attribute("href") || amountNode->Attribute("id"))
      {
          if(amountNode->Attribute("id"))
          {
             amountIDRef_ = amountNode->Attribute("id");
             amount_ = boost::shared_ptr<Money>(new Money(amountNode));
             amount_->setID(amountIDRef_);
             IDManager::instance().setID(amountIDRef_,amount_);
          }
          else if(amountNode->Attribute("href")) { amountIDRef_ = amountNode->Attribute("href");}
          else { QL_FAIL("id or href error"); }
      }
      else { amount_ = boost::shared_ptr<Money>(new Money(amountNode));}
   }

   //accruedInterestNode ----------------------------------------------------------------------------------------------------------------------
   TiXmlElement* accruedInterestNode = xmlNode->FirstChildElement("accruedInterest");

   if(accruedInterestNode){accruedInterestIsNull_ = false;}
   else{accruedInterestIsNull_ = true;}

   #ifdef ConsolePrint
      FileManager::instance().outFile_ << FileManager::instance().tap_.c_str() << "- accruedInterestNode , address : " << accruedInterestNode << std::endl;
   #endif
   if(accruedInterestNode)
   {
      if(accruedInterestNode->Attribute("href") || accruedInterestNode->Attribute("id"))
      {
          if(accruedInterestNode->Attribute("id"))
          {
             accruedInterestIDRef_ = accruedInterestNode->Attribute("id");
             accruedInterest_ = boost::shared_ptr<Money>(new Money(accruedInterestNode));
             accruedInterest_->setID(accruedInterestIDRef_);
             IDManager::instance().setID(accruedInterestIDRef_,accruedInterest_);
          }
          else if(accruedInterestNode->Attribute("href")) { accruedInterestIDRef_ = accruedInterestNode->Attribute("href");}
          else { QL_FAIL("id or href error"); }
      }
      else { accruedInterest_ = boost::shared_ptr<Money>(new Money(accruedInterestNode));}
   }

    #ifdef ConsolePrint
        FileManager::instance().tap_ = initialtap_;
    #endif 
}
boost::shared_ptr<XsdTypeDate> PendingPayment::getPaymentDate()
{
   if(!this->paymentDateIsNull_){
        if(paymentDateIDRef_ != ""){
             return boost::shared_static_cast<XsdTypeDate>(IDManager::instance().getID(paymentDateIDRef_));
        }else{
             return this->paymentDate_;
        }
   }else
   {
      QL_FAIL("null Ptr");
      return boost::shared_ptr<XsdTypeDate>();
   }
}
boost::shared_ptr<Money> PendingPayment::getAmount()
{
   if(!this->amountIsNull_){
        if(amountIDRef_ != ""){
             return boost::shared_static_cast<Money>(IDManager::instance().getID(amountIDRef_));
        }else{
             return this->amount_;
        }
   }else
   {
      QL_FAIL("null Ptr");
      return boost::shared_ptr<Money>();
   }
}
boost::shared_ptr<Money> PendingPayment::getAccruedInterest()
{
   if(!this->accruedInterestIsNull_){
        if(accruedInterestIDRef_ != ""){
             return boost::shared_static_cast<Money>(IDManager::instance().getID(accruedInterestIDRef_));
        }else{
             return this->accruedInterest_;
        }
   }else
   {
      QL_FAIL("null Ptr");
      return boost::shared_ptr<Money>();
   }
}
}

File: OTCDerivativesCalculatorModule/Project_CSharp/XmlBuilder/XmlSerializedClass/fpml-asset-5-4/PendingPayment.cs
using System;
using System.Text;
using System.Collections.Generic;
using System.Xml;

namespace FpmlSerializedCSharp
{
    public class PendingPayment : PaymentBase
    {
        public PendingPayment(XmlNode xmlNode)
        : base(xmlNode)
        {
            XmlNode paymentDateNode = xmlNode.SelectSingleNode("paymentDate");
            
            if (paymentDateNode != null)
            {
                if (paymentDateNode.Attributes["href"] != null || paymentDateNode.Attributes["id"] != null) 
                {
                    if (paymentDateNode.Attributes["id"] != null) 
                    {
                        paymentDateIDRef_ = paymentDateNode.Attributes["id"].Value;
                        XsdTypeDate ob = new XsdTypeDate(paymentDateNode);
                        IDManager.SetID(paymentDateIDRef_, ob);
                    }
                    else if (paymentDateNode.Attributes["href"] != null)
                    {
                        paymentDateIDRef_ = paymentDateNode.Attributes["href"].Value;
                    }
                    else
                    {
                        paymentDate_ = new XsdTypeDate(paymentDateNode);
                    }
                }
                else
                {
                    paymentDate_ = new XsdTypeDate(paymentDateNode);
                }
            }
            
        
            XmlNode amountNode = xmlNode.SelectSingleNode("amount");
            
            if (amountNode != null)
            {
                if (amountNode.Attributes["href"] != null || amountNode.Attributes["id"] != null) 
                {
                    if (amountNode.Attributes["id"] != null) 
                    {
                        amountIDRef_ = amountNode.Attributes["id"].Value;
                        Money ob = new Money(amountNode);
                        IDManager.SetID(amountIDRef_, ob);
                    }
                    else if (amountNode.Attributes["href"] != null)
                    {
                        amountIDRef_ = amountNode.Attributes["href"].Value;
                    }
                    else
                    {
                        amount_ = new Money(amountNode);
                    }
                }
                else
                {
                    amount_ = new Money(amountNode);
                }
            }
            
        
            XmlNode accruedInterestNode = xmlNode.SelectSingleNode("accruedInterest");
            
            if (accruedInterestNode != null)
            {
                if (accruedInterestNode.Attributes["href"] != null || accruedInterestNode.Attributes["id"] != null) 
                {
                    if (accruedInterestNode.Attributes["id"] != null) 
                    {
                        accruedInterestIDRef_ = accruedInterestNode.Attributes["id"].Value;
                        Money ob = new Money(accruedInterestNode);
                        IDManager.SetID(accruedInterestIDRef_, ob);
                    }
                    else if (accruedInterestNode.Attributes["href"] != null)
                    {
                        accruedInterestIDRef_ = accruedInterestNode.Attributes["href"].Value;
                    }
                    else
                    {
                        accruedInterest_ = new Money(accruedInterestNode);
                    }
                }
                else
                {
                    accruedInterest_ = new Money(accruedInterestNode);
                }
            }
            
        
        }
        
    
        #region PaymentDate_
        private XsdTypeDate paymentDate_;
        public XsdTypeDate PaymentDate_
        {
            get
            {
                if (this.paymentDate_ != null)
                {
                    return this.paymentDate_; 
                }
                else if (this.paymentDateIDRef_ != null)
                {
                    paymentDate_ = IDManager.getID(paymentDateIDRef_) as XsdTypeDate;
                    return this.paymentDate_; 
                }
                else
                {
                      return this.paymentDate_; 
                }
            }
            set
            {
                if (this.paymentDate_ != value)
                {
                    this.paymentDate_ = value;
                }
            }
        }
        #endregion
        
        public string paymentDateIDRef_ { get; set; }
        #region Amount_
        private Money amount_;
        public Money Amount_
        {
            get
            {
                if (this.amount_ != null)
                {
                    return this.amount_; 
                }
                else if (this.amountIDRef_ != null)
                {
                    amount_ = IDManager.getID(amountIDRef_) as Money;
                    return this.amount_; 
                }
                else
                {
                      return this.amount_; 
                }
            }
            set
            {
                if (this.amount_ != value)
                {
                    this.amount_ = value;
                }
            }
        }
        #endregion
        
        public string amountIDRef_ { get; set; }
        #region AccruedInterest_
        private Money accruedInterest_;
        public Money AccruedInterest_
        {
            get
            {
                if (this.accruedInterest_ != null)
                {
                    return this.accruedInterest_; 
                }
                else if (this.accruedInterestIDRef_ != null)
                {
                    accruedInterest_ = IDManager.getID(accruedInterestIDRef_) as Money;
                    return this.accruedInterest_; 
                }
                else
                {
                      return this.accruedInterest_; 
                }
            }
            set
            {
                if (this.accruedInterest_ != value)
                {
                    this.accruedInterest_ = value;
                }
            }
        }
        #endregion
        
        public string accruedInterestIDRef_ { get; set; }
        
    
        
    
    }
    
}

